plugins/sub_plugins/validate/jsonschema.py

```diff
@@ -153,30 +207,43 @@ def _validate_jsonschema(self):
         error_messages = None
 
         draft = self._get_sub_plugin_options("draft")
+        check_format = self._get_sub_plugin_options("check_format")
         error_messages = []
 
         for criteria in self._criteria:
-            if draft == "draft3":
-                validator = jsonschema.Draft3Validator(
-                    criteria,
-                    format_checker=jsonschema.draft3_format_checker,
-                )
-            elif draft == "draft4":
-                validator = jsonschema.Draft4Validator(
-                    criteria,
-                    format_checker=jsonschema.draft4_format_checker,
-                )
-            elif draft == "draft6":
-                validator = jsonschema.Draft6Validator(
-                    criteria,
-                    format_checker=jsonschema.draft6_format_checker,
+            format_checker = None
+            validator_class = None
+            try:
+                validator_class = self._JSONSCHEMA_DRAFTS[draft]["validator"]
+            except KeyError:
+                display.vvv(
+                    'No jsonschema validator for "{draft}", falling back to autodetection.'.format(
+                        draft=draft,
+                    ),
                 )
-            else:
-                validator = jsonschema.Draft7Validator(
-                    criteria,
-                    format_checker=jsonschema.draft7_format_checker,
-                )
-
+            if validator_class is None:
+                # Either no draft was specified or specified draft has no validator class
+                # in installed jsonschema version. Do autodetection instead.
+                validator_class = jsonschema.validators.validator_for(criteria)
+
+            if check_format:
+                try:
+                    format_checker = validator_class.FORMAT_CHECKER
+                except AttributeError:
+                    # On older jsonschema versions there is no connection between a validator and the correct format
```

Comment:
I don't know if this is the version that added it, but having some hints about when we can remove certain things helps get them removed when they are no longer relevant
```suggestion
                    #TODO: Remove when Python 3.6 support is dropped.
                    # On jsonschema<4.5, there is no connection between a validator and the correct format
```